networkx/algorithms/connectivity/edge_kcomponents.py

```diff
@@ -0,0 +1,570 @@
+        than O(|V|).
+
+        For single values of k it is faster to use `nx.k_edge_subgraphs`.
+        But for multiple values of k, it can be faster to build AuxGraph and
+        then use this method.
+        """
+        if k < 1:
+            raise ValueError('k cannot be less than 1')
+        H = self.H
+        A = self.A
+        # "traverse the auxiliary graph A and delete all edges with weights less
+        # than k"
+        aux_weights = nx.get_edge_attributes(A, 'weight')
+        # Create a relevant graph with the auxillary edges with weights >= k
+        R = nx.Graph()
+        R.add_nodes_from(A.nodes())
+        R.add_edges_from(e for e, w in aux_weights.items() if w >= k)
+
+        # Return the components whose subgraphs are k-edge-connected
+        for cc in nx.connected_components(R):
+            if len(cc) < k:
+                # Early return optimization
+                for node in cc:
+                    yield {node}
+            else:
+                # Call subgraph solution to refine the results
+                C = H.subgraph(cc)
+                for sub_cc in k_edge_subgraphs(C, k):
+                    yield sub_cc
+
+
+def _low_degree_nodes(G, k, nbunch=None):
+    """ Helper. Find nodes with degree less than k """
+    # Nodes with degree less than k cannot be k-edge-connected.
+    if G.is_directed():
+        # Consider both in and out degree in the directed case
+        seen = set()
+        for node, degree in G.out_degree(nbunch):
+            if degree < k:
+                seen.add(node)
+                yield node
+        for node, degree in G.in_degree(nbunch):
+            if node not in seen and degree < k:
+                seen.add(node)
+                yield node
+    else:
+        # Only the degree matters in the undirected case
+        for node, degree in G.degree(nbunch):
+            if degree < k:
+                yield node
+
+
+def _high_degree_components(G, k):
+    """
+    Helper. Removes and generates each node with degree less than k.  Then
+    generates remaining components where all nodes have degree at least k.
+    """
+    # Iteravely remove parts of the graph that are not k-edge-connected
+    H = G.copy()
+    singletons = set(_low_degree_nodes(H, k))
+    while singletons:
+        # Only search neighbors of removed nodes
+        nbunch = set(it.chain.from_iterable(map(H.neighbors, singletons)))
+        nbunch.difference_update(singletons)
+        H.remove_nodes_from(singletons)
+        for node in singletons:
+            yield {node}
+        singletons = set(_low_degree_nodes(H, k, nbunch))
+
+    # Note: remaining connected compoments may not be k-edge-connected
+    if G.is_directed():
+        for cc in nx.strongly_connected_components(H):
+            yield cc
+    else:
+        for cc in nx.connected_components(H):
+            yield cc
+
+
+def general_k_edge_subgraphs(G, k):
+    """
```

Comment:
This function should probably be renamed or removed from `__all__`, but I'm sure what the best name for this function is. Currently I have it in `__all__` because I want to use it in the tests. I can't seem to access it otherwise (is there a way I can get to the function even if its not in `__all__`?).